### DISCUSSION TITLE
API examples / third party migrators

### DISCUSSION THREAD

**Discussion by chadrik** [outline/outline#4981] [General]
I'd liked to write a tool from to migrate from our internal wiki to Outline.  I've started poking around the Outline code itself, but it would be really useful to have a compact example, such as a migration tool for another wiki.   For example, some things that aren't clear from reading the API docs are how to upload an image after creating an Attachment, and how to reorder sibling Documents.  Are there any open source third party migrators/tools out there for Outline that would be good reference?  

thanks in advance.

**Comment by tracure1337** [comment]
I am also interested in this and could not find much so far.
Currently migrating everything by hand which is a great way to cleanup and improve the quality though ;-)

**Reply by chadrik** [reply]
I found this third party tool that can do migrations and git sync, but images are currently not imported: https://github.com/captn3m0/outliner

**Comment by Limezy** [comment]
If I'm not wrong there is now a json import / export format for Outline, that you may use as a basis for your migration tool ? You could also have a look into the various built-in imports in the codebase (although I don't think they use this same json format)
